### PR TITLE
Remove muon importer

### PR DESCRIPTION
**Pull request by simonhong**
Resolves https://github.com/brave/brave-browser/issues/7403

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [x] Android
  - [ ] iOS
  - [x] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: browser/importer/brave_profile_writer.cc [hunk 1]
```diff
@@ -1,519 +0,0 @@
...
-      site->id = publisher.key;
-      site->status = 0;
-      site->excluded = 0;
-      site->name = publisher.name;
-      site->favicon_url = "";
-      site->url = publisher.url;
-      site->provider = publisher.provider;
-
-      // Add `recurring_donation` entry
-      rewards_service->OnTip(publisher.key,
```
Comment (NejcZdovc) [inline browser/importer/brave_profile_writer.cc]: based on the fact that this one is deleted @simonhong could you please remove `OnTip` function from rewards_service that accepts this params as it was only kept for this use case. Thank you

Comment (simonhong) [reply browser/importer/brave_profile_writer.cc]: There are one more user of that api - https://github.com/brave/brave-core/blob/master/components/brave_rewards/browser/rewards_service_browsertest.cc#L1469

Comment (simonhong) [reply browser/importer/brave_profile_writer.cc]: @NejcZdovc I tried to cleanup test files also with below patch. but got many test failure.
I think this cleanup should be done as a f/u.

```
diff --git a/components/brave_rewards/browser/rewards_service.h b/components/brave_rewards/browser/rewards_service.h
index 14c36c1c0..5091f022b 100644
--- a/components/brave_rewards/browser/rewards_service.h
+++ b/components/brave_rewards/browser/rewards_service.h
@@ -206,13 +206,6 @@ class RewardsService : public KeyedService {
       const double amount,
       const bool recurring) = 0;
 
-  // Used in importer from muon days
-  virtual void OnTip(
-      const std::string& publisher_key,
-      double amount,
-      const bool recurring,
-      std::unique_ptr<brave_rewards::ContentSite> site) = 0;
-
   virtual void RemoveRecurringTipUI(const std::string& publisher_key) = 0;
   virtual void GetRecurringTipsUI(GetRecurringTipsCallback callback) = 0;
   virtual void GetOneTimeTipsUI(GetOneTimeTipsCallback callback) = 0;
diff --git a/components/brave_rewards/browser/rewards_service_browsertest.cc b/components/brave_rewards/browser/rewards_service_browsertest.cc
index effd11e7e..213b97550 100644
--- a/components/brave_rewards/browser/rewards_service_browsertest.cc
+++ b/components/brave_rewards/browser/rewards_service_browsertest.cc
@@ -1453,20 +1453,12 @@ class BraveRewardsBrowserTest
   void TipViaCode(
       const std::string publisher_key,
       const double amount,
-      const ledger::PublisherStatus status,
       const bool should_contribute = false,
       const bool recurring = false,
       const ledger::Result result = ledger::Result::LEDGER_OK) {
     tip_reconcile_completed_ = false;
     pending_tip_saved_ = false;
-    auto site = std::make_unique<brave_rewards::ContentSite>();
-    site->id = publisher_key;
-    site->name = publisher_key;
-    site->url = publisher_key;
-    site->status = static_cast<int>(status);
-    site->provider = "";
-    site->favicon_url = "";
-    rewards_service_->OnTip(publisher_key, amount, recurring, std::move(site));
+    rewards_service_->OnTip(publisher_key, amount, recurring);
 
     if (recurring) {
       return;
@@ -2388,14 +2380,12 @@ IN_PROC_BROWSER_TEST_F(BraveRewardsBrowserTest,
   TipViaCode(
       "duckduckgo.com",
       20.0,
-      ledger::PublisherStatus::VERIFIED,
       false,
       true);
 
   TipViaCode(
       "brave.com",
       50.0,
-      ledger::PublisherStatus::NOT_VERIFIED,
       false,
       true);
 
@@ -2426,14 +2416,12 @@ IN_PROC_BROWSER_TEST_F(BraveRewardsBrowserTest,
   TipViaCode(
       "duckduckgo.com",
       50.0,
-      ledger::PublisherStatus::VERIFIED,
       false,
       true);
 
   TipViaCode(
       "brave.com",
       50.0,
-      ledger::PublisherStatus::NOT_VERIFIED,
       false,
       true);
 
@@ -2476,12 +2464,12 @@ IN_PROC_BROWSER_TEST_F(BraveRewardsBrowserTest,
   EXPECT_TRUE(WaitForLoadStop(contents()));
 
   // Tip unverified publisher
-  TipViaCode("brave.com", 1.0, ledger::PublisherStatus::NOT_VERIFIED);
-  TipViaCode("brave.com", 5.0, ledger::PublisherStatus::NOT_VERIFIED);
-  TipViaCode("3zsistemi.si", 10.0, ledger::PublisherStatus::NOT_VERIFIED);
-  TipViaCode("3zsistemi.si", 5.0, ledger::PublisherStatus::NOT_VERIFIED);
-  TipViaCode("3zsistemi.si", 10.0, ledger::PublisherStatus::NOT_VERIFIED);
-  TipViaCode("3zsistemi.si", 10.0, ledger::PublisherStatus::NOT_VERIFIED);
+  TipViaCode("brave.com", 1.0);
+  TipViaCode("brave.com", 5.0);
+  TipViaCode("3zsistemi.si", 10.0);
+  TipViaCode("3zsistemi.si", 5.0);
+  TipViaCode("3zsistemi.si", 10.0);
+  TipViaCode("3zsistemi.si", 10.0);
 
   // Claim promotion using panel
   ClaimPromotion(true);
@@ -2568,7 +2556,6 @@ IN_PROC_BROWSER_TEST_F(BraveRewardsBrowserTest, RewardsPanelDefaultTipChoices) {
   TipViaCode(
       "3zsistemi.si",
       10.0,
-      ledger::PublisherStatus::VERIFIED,
       false,
       true);
 
@@ -2679,7 +2666,6 @@ IN_PROC_BROWSER_TEST_F(BraveRewardsBrowserTest,
   TipViaCode(
       "duckduckgo.com",
       amount,
-      ledger::PublisherStatus::VERIFIED,
       should_contribute);
   VerifyTip(amount, should_contribute, false, true);
 
@@ -2712,14 +2698,12 @@ IN_PROC_BROWSER_TEST_F(BraveRewardsBrowserTest,
   TipViaCode(
       "duckduckgo.com",
       amount,
-      ledger::PublisherStatus::VERIFIED,
       should_contribute);
   total_amount += amount;
 
   TipViaCode(
       "laurenwags.github.io",
       amount,
-      ledger::PublisherStatus::VERIFIED,
       should_contribute);
   total_amount += amount;
 
@@ -2752,7 +2736,6 @@ IN_PROC_BROWSER_TEST_F(BraveRewardsBrowserTest, TipConnectedPublisherAnon) {
   TipViaCode(
       "bumpsmack.com",
       amount,
-      ledger::PublisherStatus::CONNECTED,
       should_contribute);
   VerifyTip(amount, should_contribute, false, true);
 }
@@ -2785,7 +2768,6 @@ IN_PROC_BROWSER_TEST_F(
   TipViaCode(
       "bumpsmack.com",
       amount,
-      ledger::PublisherStatus::CONNECTED,
       should_contribute);
   VerifyTip(amount, should_contribute, false, true);
 }
@@ -2816,7 +2798,6 @@ IN_PROC_BROWSER_TEST_F(
   TipViaCode(
       "bumpsmack.com",
       amount,
-      ledger::PublisherStatus::CONNECTED,
       should_contribute,
       false,
       ledger::Result::LEDGER_ERROR);
@@ -2850,7 +2831,6 @@ IN_PROC_BROWSER_TEST_F(
   TipViaCode(
       "bumpsmack.com",
       amount,
-      ledger::PublisherStatus::CONNECTED,
       should_contribute,
       false,
       ledger::Result::LEDGER_ERROR);
@@ -2908,7 +2888,6 @@ IN_PROC_BROWSER_TEST_F(BraveRewardsBrowserTest,
   TipViaCode(
       "duckduckgo.com",
       25.0,
-      ledger::PublisherStatus::VERIFIED,
       false,
       true);
 
@@ -2946,28 +2925,24 @@ IN_PROC_BROWSER_TEST_F(BraveRewardsBrowserTest,
   TipViaCode(
       "duckduckgo.com",
       5.0,
-      ledger::PublisherStatus::VERIFIED,
       false,
       true);
 
   TipViaCode(
       "site1.com",
       10.0,
-      ledger::PublisherStatus::VERIFIED,
       false,
       true);
 
   TipViaCode(
       "site2.com",
       10.0,
-      ledger::PublisherStatus::VERIFIED,
       false,
       true);
 
   TipViaCode(
       "site3.com",
       10.0,
-      ledger::PublisherStatus::VERIFIED,
       false,
       true);
 
diff --git a/components/brave_rewards/browser/rewards_service_impl.cc b/components/brave_rewards/browser/rewards_service_impl.cc
index efc5947f1..57915102e 100644
--- a/components/brave_rewards/browser/rewards_service_impl.cc
+++ b/components/brave_rewards/browser/rewards_service_impl.cc
@@ -2513,31 +2513,6 @@ void RewardsServiceImpl::GetRewardsInternalsInfo(
                      AsWeakPtr(), std::move(callback)));
 }
 
-void RewardsServiceImpl::OnTip(
-    const std::string& publisher_key,
-    const double amount,
-    const bool recurring,
-    std::unique_ptr<brave_rewards::ContentSite> site) {
-  if (!site) {
-    return;
-  }
-
-  auto info = ledger::PublisherInfo::New();
-  info->id = publisher_key;
-  info->name = site->name;
-  info->url = site->url;
-  info->provider = site->provider;
-  info->favicon_url = site->favicon_url;
-
-  bat_ledger_->SavePublisherInfo(
-      std::move(info),
-      base::BindOnce(&RewardsServiceImpl::OnTipPublisherSaved,
-          AsWeakPtr(),
-          publisher_key,
-          amount,
-          recurring));
-}
-
 void RewardsServiceImpl::OnTipPublisherSaved(
     const std::string& publisher_key,
     const double amount,
diff --git a/components/brave_rewards/browser/rewards_service_impl.h b/components/brave_rewards/browser/rewards_service_impl.h
index f152aabe8..c877fa28c 100644
--- a/components/brave_rewards/browser/rewards_service_impl.h
+++ b/components/brave_rewards/browser/rewards_service_impl.h
@@ -247,12 +247,6 @@ class RewardsServiceImpl : public RewardsService,
 
   void RemoveAllPendingContributionsUI() override;
 
-  void OnTip(
-      const std::string& publisher_key,
-      const double amount,
-      const bool recurring,
-      std::unique_ptr<brave_rewards::ContentSite> site) override;
-
   void OnTip(
       const std::string& publisher_key,
       const double amount,

```

Comment (NejcZdovc) [reply browser/importer/brave_profile_writer.cc]: hi, sorry forgot about browser tests. Thanks for trying, let's keep it as is. Sorry for a noice

Comment (simonhong) [reply browser/importer/brave_profile_writer.cc]: NP :)

**Comment by simonhong** [comment]
@NejcZdovc Can you check `RewardsService::CheckImported()` is also deprecated at https://github.com/brave/brave-core/blob/master/components/brave_rewards/browser/rewards_service.h#L223?

**Comment by NejcZdovc** [comment]
@simonhong yup that is deprecated. So we can remove this function, then `kBravePaymentsPinnedItemCount` and all flow in TS that calls `CheckImported` through brave_page_ui

**Comment by simonhong** [comment]
@NejcZdovc Thanks for checking. I found this method while trying to remove `KBravePaymentsPinnedItemCount` pref. :)

**Comment by simonhong** [comment]
Rebased on C81.

**Comment by simonhong** [comment]
NOTE: Delete muon import types after merging https://github.com/brave/brave-core/pull/5018

**Comment by simonhong** [comment]
handled https://github.com/brave/brave-core/pull/5018

**Comment by simonhong** [comment]
@NejcZdovc Can I get owner review?

**Pull request merged**